### DISCUSSION TITLE
How do I test that an endpoint *shouldn't* work?

### DISCUSSION THREAD

**Discussion by tamoreton** [usebruno/bruno#909] [Q&A]
Say I have an API secured with basic auth and some credentials, set as environment variables. I want to test that an endpoint works when credentials are provided, and that it doesn't work when credentials are not provided.

How do I do this? I've tried a few approaches and none have had the results that I expected (actually, I was struggling to reason about some of them – I didn't find the [JavaScript API Reference](https://docs.usebruno.com/scripting/javascript-reference.html) very clear).

I've tried:

```javascript
test("should be ok", () => {
  expect(res.getStatus()).to.equal(200);
});

req.setHeader("authorization", "");

test("should prevent unauthorised access", () => {
  expect(res.getStatus()).to.equal(401);
});
```

and

```javascript
test("should be ok", () => {
  expect(res.getStatus()).to.equal(200);
});

test("should prevent unauthorised access", () => {
  bru.setEnvVar("password", "wrongpass);
  expect(res.getStatus()).to.equal(401);
  bru.setEnvVar("password", "password");
});
```

**Comment by Sl-Alex** [comment]
In this case I would just create two requests: one with valid auth data (and check status 200), and one without (and check for 401)
There is a possibility to call some additional endpoint in a pre- or post-request script (see [here](https://docs.usebruno.com/scripting/sync-requests.html)), but IMHO it is much easier (and better) to create an additional request.

For example, this is how I test the very same endpoint:
![image](https://github.com/usebruno/bruno/assets/7002691/42d5c9f3-9b5a-4259-a1c9-ee6be68a05d5)
